tests/test_api.py

```diff
@@ -82,9 +82,12 @@ def test_representations(algorithm):
 @pytest.mark.parametrize("allocation_algorithm", available_allocation_algorithms)
 def test_calculation_positions(test_df, allocation_algorithm):
     """Checks algorithms calculate positions and returns."""
-
+    test_df_copy = test_df.copy()
```

Comment:
would it not create a shallow copy?  why are we doing this?

Reply:
Doing this to avoid making changes to test_df. Doing this actually solved a bug.
deep = True by default

Reply:
gotcha ! Make sense. So we are actually deep copying.